tiledb/sm/subarray/subarray.h

```diff
@@ -247,9 +247,28 @@ class Subarray {
    */
   const Config* config() const;
 
-  /** equivalent for older Query::set_subarray(const void *subarray); */
+  /**
+   * Sets the subarray using a pointer to raw range data that stores one range
+   * per dimension.
+   *
+   * This is only valid for arrays with homogenous dimension data types.
+   *
+   * @param subarray A pointer to the range data to use.
+   * @returns Status error
+   */
   Status set_subarray(const void* subarray);
 
+  /**
+   * Sets the subarray using a pointer to raw range data that stores one range
+   * per dimension without performing validity checks.
+   *
+   * This is only valid for arrays with homogenous dimension data types. This
```

Comment:
Not a big deal but the comment should say This function should only be used for deser...